Use pre-calculated value in trace message

lukezaparaniuk opened the pull request:
Replaced the math.min(size, sizeInBytes) call with the count variable that has already been calculated within the writeTo method

lukezaparaniuk committed:
Use pre-calculated value in trace message

Replaced the math.min(size, sizeInBytes) call with the count variable that has already been calculated within the writeTo method

lukezaparaniuk committed:
Use pre-calculated value in trace message

Replaced the math.min(size, sizeInBytes) call with the count variable that has already been calculated within the writeTo method

nepal committed:
KAFKA-2063; Bound fetch response size (KIP-74)

This PR is implementation of [KIP-74](https://cwiki.apache.org/confluence/display/KAFKA/KIP-74%3A+Add+Fetch+Response+Size+Limit+in+Bytes) which is originally motivated by [KAFKA-2063](https://issues.apache.org/jira/browse/KAFKA-2063).

Author: Andrey Neporada <neporada@gmail.com>
Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>, Jiangjie Qin <becket.qin@gmail.com>, Jason Gustafson <jason@confluent.io>

Closes #1812 from nepal/kip-74

ijuma commented on core/src/main/scala/kafka/log/FileMessageSet.scala:
```diff
@@ -241,7 +241,7 @@ class FileMessageSet private[kafka](@volatile var file: File,
     }).toInt
     trace("FileMessageSet " + file.getAbsolutePath + " : bytes transferred : " + bytesTransferred
-      + " bytes requested for transfer : " + math.min(size, sizeInBytes))
+      + " bytes requested for transfer : " + count)
```
While you're at it, can you please replace string concatenation with string interpolation (http://docs.scala-lang.org/overviews/core/string-interpolation.html)?

lukezaparaniuk committed:
MINOR: Fix reference to argument in `LogSegment.translateOffset`

Changed the lowerBound argument reference in the summary comment of the translateOffset method to match the actual argument name: startingFilePosition.

Author: Luke Zaparaniuk <luke.zaparaniuk@gmail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>

Closes #1876 from lukezaparaniuk/patch-1

dguy committed:
HOTFIX: logic in QuerybaleStateIntegrationTest.shouldBeAbleToQueryState incorrect

The logic in `verifyCanGetByKey` was incorrect. It was
```
windowState.size() < keys.length &&
countState.size() < keys.length &&
System.currentTimeMillis() < timeout
```
but should be:
```
(windowState.size() < keys.length || countState.size() < keys.length) && System.currentTimeMillis() < timeout
```

Author: Damian Guy <damian.guy@gmail.com>

Reviewers: Guozhang Wang <wangguoz@gmail.com>

Closes #1879 from dguy/minor-fix-test

dguy committed:
KAFKA-4175: Can't have StandbyTasks in KafkaStreams where NUM_STREAM_THREADS_CONFIG > 1

standby tasks should be assigned per consumer not per process

Author: Damian Guy <damian.guy@gmail.com>

Reviewers: Eno Thereska, Guozhang Wang

Closes #1862 from dguy/kafka-4175

dguy committed:
KAFKA-4163: NPE in StreamsMetadataState during re-balance operations

During rebalance operations the Cluster object gets set to Cluster.empty(). This can result in NPEs when doing certain operation on StreamsMetadataState. This should throw a StreamsException if the Cluster is empty as it is not yet (re-)initialized

Author: Damian Guy <damian.guy@gmail.com>

Reviewers: Eno Thereska, Guozhang Wang

Closes #1845 from dguy/streams-meta-hotfix

rajinisivaram committed:
KAFKA-4079; Documentation for secure quotas

Details in KIP-55.

Author: Rajini Sivaram <rajinisivaram@googlemail.com>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #1847 from rajinisivaram/KAFKA-4079

lukezaparaniuk committed:
Merge branch 'patch-2' of https://github.com/lukezaparaniuk/kafka into patch-2
